public/pages/Snapshots/components/RestoreActivitiesPanel/RestoreActivitiesPanel.tsx

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiInMemoryTable, EuiSpacer, EuiLink, EuiFlyout, EuiButton } from "@elastic/eui";
+import _ from "lodash";
+import React, { useEffect, useContext, useState } from "react";
+import { SnapshotManagementService } from "../../../../services";
+import { CoreServicesContext } from "../../../../components/core_services";
+import { getErrorMessage } from "../../../../utils/helpers";
+import { GetIndexRecoveryResponse, CatSnapshotIndex } from "../../../../../server/models/interfaces";
+import { BREADCRUMBS } from "../../../../utils/constants";
+import { ContentPanel } from "../../../../components/ContentPanel";
+
+interface RestoreActivitiesPanelProps {
+  snapshotManagementService: SnapshotManagementService;
+  snapshotId: string;
+  repository: string;
+}
+
+export const RestoreActivitiesPanel = ({ snapshotManagementService, snapshotId }: RestoreActivitiesPanelProps) => {
+  const context = useContext(CoreServicesContext);
+  const [startTime, setStartTime] = useState("");
+  const [stopTime, setStopTime] = useState("");
+  const [stage, setStage] = useState("");
+  const [indices, setIndices] = useState([{}]);
+  const [flyout, setFlyout] = useState(false);
+
+  useEffect(() => {
+    context!.chrome.setBreadcrumbs([BREADCRUMBS.SNAPSHOT_MANAGEMENT, BREADCRUMBS.SNAPSHOTS, BREADCRUMBS.SNAPSHOT_RESTORE]);
+    getRestoreStatus();
+  }, []);
+
+  const getRestoreStatus = async () => {
+    if (stage.indexOf("DONE") >= 0) {
+      console.log("done");
+      return;
+    }
+    try {
+      const res = await snapshotManagementService.getIndexRecovery();
+
+      if (res.ok) {
+        const response: GetIndexRecoveryResponse = res.response;
+
+        setRestoreStatus(response);
+        console.log(response);
+      } else {
+        context?.notifications.toasts.addDanger(res.error);
+      }
+    } catch (err) {
+      context?.notifications.toasts.addDanger(getErrorMessage(err, "There was a problem loading the recovery."));
+    }
+  };
+
+  const onIndexesClick = (e: React.MouseEvent) => {
+    e.preventDefault();
+    setFlyout(true);
```

Comment:
flyout will be in the next PR?

Reply:
Flyout is on line 143, no content at this stage.  that comes in later PR